hasher-matcher-actioner/terraform/main.tf

```diff
@@ -242,6 +242,154 @@ module "authentication" {
+ * complexity can't be handled by this "base" lambda, it will be routed to
+ * another specially capable lambda queue.
+ *
+ * - This should soon absorb the pdq_images queue + SNS topic as the only queue
+ *   that we will publish submissions on.
+ *   If we have proven that the generic lambda can generate PDQ signals, we can 
+ *   do away with the PDQ specific infrastructure altogether.
+*/
+resource "aws_sqs_queue" "submissions_queue" {
+  name_prefix                = "${var.prefix}-submissions"
+  visibility_timeout_seconds = 300
+  message_retention_seconds  = 1209600
+  tags = merge(
+    var.additional_tags,
+    local.common_tags,
+    {
+      Name = "SubmissionsQueue"
+    }
+  )
+}
+
+
+/* Hashing Lambda Configuration Begins
+ *
+ * Pipe the submissions queue into the hashing lambda.
+ */
+resource "aws_lambda_function" "hashing_lambda" {
+  function_name = "${var.prefix}_hasher"
+  package_type  = "Image"
+  role          = aws_iam_role.hashing_lambda_role.arn
+  image_uri     = var.hma_lambda_docker_uri
+  image_config {
+    command = ["hmalib.lambdas.hashing.lambda_handler"]
+  }
+
+  timeout     = 300
+  memory_size = 512
+
+  environment {
+    variables = {
+      DYNAMODB_TABLE      = module.datastore.primary_datastore.name
+      METRICS_NAMESPACE   = var.metrics_namespace
+      MEASURE_PERFORMANCE = var.measure_performance ? "True" : "False"
+      # HASHES_QUEUE_URL    = module.pdq_signals.hashes_queue_url
+    }
+  }
+
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "HashingLambda"
+    }
+  )
+}
+
+data "aws_iam_policy_document" "lambda_assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_policy" "hashing_lambda_policy" {
+  name_prefix = "${var.prefix}_hasher_policy"
+  description = "Permissions for Hashing Lambda"
+  policy      = data.aws_iam_policy_document.hashing_lambda.json
+}
+resource "aws_iam_role" "hashing_lambda_role" {
+  name_prefix        = "${var.prefix}_hasher"
+  assume_role_policy = data.aws_iam_policy_document.lambda_assume_role.json
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "HashingLambda"
+    }
+  )
+}
+
+resource "aws_cloudwatch_log_group" "hashing_logs" {
+  name              = "/aws/lambda/${aws_lambda_function.hashing_lambda.function_name}"
+  retention_in_days = var.log_retention_in_days
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "HasherLambdaLogGroup"
+    }
+  )
+}
+
+data "aws_iam_policy_document" "hashing_lambda" {
+  statement {
+    effect    = "Allow"
+    actions   = ["sqs:GetQueueAttributes", "sqs:ReceiveMessage", "sqs:DeleteMessage"]
+    resources = [aws_sqs_queue.submissions_queue.arn]
+  }
+  /*statement {
+    effect    = "Allow"
+    actions   = ["sqs:SendMessage"]
+    resources = [aws_sqs_queue.hashes_queue.arn]
+  }*/
```

Comment:
remove?

Reply:
will be required once hashes are published.